library/src/lib/select/select.component.ts

```diff
@@ -0,0 +1,395 @@
+
+    /** @hidden */
+    @HostBinding('class.fd-dropdown')
+    fdDropdownClass: boolean = true;
+
+    /** @hidden */
+    @ContentChildren(OptionComponent, { descendants: true })
+    options: QueryList<OptionComponent>;
+
+    /** Whether the select component is disabled. */
+    @Input()
+    disabled: boolean = false;
+
+    /** Placeholder for the select. Appears in the triggerbox if no option is selected. */
+    @Input()
+    placeholder: string;
+
+    /** Open state of the select. */
+    @Input()
+    isOpen: boolean = false;
+
+    /** Current value of the selected option. */
+    @Input()
+    value: any;
+
+    /** Popper.js options of the popover. */
+    @Input()
+    popperOptions: PopperOptions = {
+        placement: 'bottom-start',
+        modifiers: {
+            preventOverflow: {
+                enabled: true,
+                escapeWithReference: true,
+                boundariesElement: 'scrollParent'
+            }
+        }
+    };
+
+    /**
+     * Preset options for the popover body width.
+     * * `at-least` will apply a minimum width to the body equivalent to the width of the control.
+     * * `equal` will apply a width to the body equivalent to the width of the control.
+     * * Leave blank for no effect.
+     */
+    @Input()
+    fillControlMode: PopoverFillMode = 'at-least';
+
+    /** Template with which to display the trigger box. */
+    @Input()
+    triggerTemplate: TemplateRef<any>;
+
+    /** The element to which the popover should be appended. */
+    @Input()
+    appendTo: HTMLElement | 'body';
+
+    /** Event emitted when the popover open state changes. */
+    @Output()
+    readonly isOpenChange: EventEmitter<boolean>
+        = new EventEmitter<boolean>();
+
+    /** Event emitted when the selected value of the select changes. */
+    @Output()
+    readonly valueChange: EventEmitter<any>
+        = new EventEmitter<any>();
+
+    /** Current selected option component reference. */
+    private selected: OptionComponent;
+
+    /** Subject triggered when the component is destroyed. */
+    private readonly destroy$ = new Subject<void>();
+
+    /** Observable triggered when an option has its selectedChange event fire. */
+    private readonly optionsStatusChanges: Observable<OptionComponent> = defer(() => {
+        const options = this.options;
+        if (options) {
+            return options.changes.pipe(
+                startWith(options),
+                switchMap(() => merge(...options.map(option => option.selectedChange)))
+            );
+        }
+    }) as Observable<OptionComponent>;
+
+    /** @hidden */
+    onChange: Function = () => {};
+
+    /** @hidden */
+    onTouched: Function = () => {};
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.value) {
+            setTimeout(() => {
+                if (this.value) {
+                    this.selectValue(this.value, false);
+                }
+            });
```

Comment:
It's to avoid the famous `ExpressionChangedAfterItHasBeenChecked`. This is actually (surprisingly) a perfectly valid way of solving it if you're inside `ngOnChanges`. The other solution would be to use a promise to defer the execution.